x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/ResourcePrivilegesMap.java

```diff
@@ -20,25 +19,19 @@
  */
 public final class ResourcePrivilegesMap {
 
-    private final boolean allAllowed;
     private final Map<String, ResourcePrivileges> resourceToResourcePrivileges;
 
-    public ResourcePrivilegesMap(boolean allAllowed, Map<String, ResourcePrivileges> resToResPriv) {
-        this.allAllowed = allAllowed;
+    public ResourcePrivilegesMap(Map<String, ResourcePrivileges> resToResPriv) {
         this.resourceToResourcePrivileges = Collections.unmodifiableMap(Objects.requireNonNull(resToResPriv));
     }
 
-    public boolean allAllowed() {
-        return allAllowed;
-    }
```

Comment:
Part of the ethos of this change was to keep the `ResourcePrivilegesMap` reserved for the check details only, so not include the authorization result (because they can exist separately). The `allAllowed` result can be confounded with the authorization result. Overall, I think it is better we don't expose it.